source/contributed/typescript-and-screeps-1.md

```diff
@@ -0,0 +1,207 @@
+
+For a quick-start, skip down to [screeps-typescript-starter](#screeps-typescript-starter).
+
+## Why TypeScript?
+
+TypeScript is a superset of ECMAScript (a fancy way to say Javascript).  That means it has *all* the features of ECMA, *plus* some extra ones.  When a project is compiled, Typescript will process all your files, and "transpile" them into the target Javascript version (ES6, in the case of Screeps).  This works similarly to Babel, if you have used that in the past.
+
+Using TypeScript can be frustrating when you first start, because it seems the compiler never stops complaining. Power through it. This ends up being the most useful feature of TypeScript once you get going. Towards then, we'll look at some concrete examples of this.
+
+This guide won't attempt to teach you typescript ([Typescript Deep Dive](https://basarat.gitbooks.io/typescript/docs/why-typescript.html) is an execellent starting resource for that), but will provide an introduction in getting set up, and using the more advanced features to your advantage
+
+## Getting Started with TypeScript
+
+> **Note**: If you already have a Typescript project that compiles, and understand the basic tooling, you can most likely skip the rest of this article
+
+This section will only provide a very brief overview of getting started with TypeScript.  For more in-depth, please see the above Deep Dive link, or the TypeScript docs.
+
+### Install Your Tooling
+
+To get off the ground with typescript, you're going to need quite a few different tools:
+
+* An IDE that supports TS ([Slant](https://www.slant.co/topics/5815/~ides-for-typescript-development) recommends Visual Studio Code as the most popular choice)
+
+* NPM (not in the scope of this article)
+
+* The typescript compiler:
+  ```bash
+  # for cutting edge in development version (currently 2.4.0):
+  $ npm install -g typescript@next
+  # for latest stable version (currently 2.3.2):
+  $ npm install -g typescript@latest
+  ```
+* [Typings](https://github.com/typings/typings) TypeScript definition manager
+  ```bash
+  $ npm install -g typings
+  ```
+* *Recommended*: [webpack](https://webpack.js.org/guides/webpack-and-typescript/)
+for bundling your project into one output file
+  ```bash
+  $ npm install -g webpack
+  ```
+* *Recommended* (but configuration not covered here): [tslint](https://palantir.github.io/tslint/)
+for keeping your coding style standard
+  ```bash
+  $ npm install -g tslint
+  ```
+
+### Creating Your First New Project
+> **Note**: This is just a brief guide to getting a bare project going. For most users, the **screeps-typescript-starter** below will be a better & more full-feature choice
+
+> **Note**: Your IDE may complete these steps for you when starting a new project... If so, go ahead and skip down to **Typings**
+
+To get a bare project going, we have to first `init` all our tools.  `cd` into your project directory (presumably, an empty folder), and run the following:
+
+```bash
+$ cd my_proj/
```

Comment:
I'll debate that generally projects include the commands they just told you to run

That said, I agree it would read better as
> `cd` into your project directory (presumably, an empty folder), and initialize all your project tools: